src/AWS.Logger.Core/Core/AWSLoggerCore.cs

```diff
@@ -122,27 +126,12 @@ public void AddMessage(string message)
             {
                 message = EMPTY_MESSAGE;
             }
-            if (_pendingMessageQueue.Count > _config.MaxQueuedMessages)
-            {
-                if ((_maxBufferTimeStamp == DateTime.MinValue) || (DateTime.Now > _maxBufferTimeStamp.Add(TimeSpan.FromMinutes(MAX_BUFFER_TIMEDIFF))))
-                {
-                    _maxBufferTimeStamp = DateTime.Now;
-                    message = "The AWS Logger in-memory buffer has reached maximum capacity";
-                    _pendingMessageQueue.Enqueue(new InputLogEvent
-                    {
-                        Timestamp = DateTime.Now,
-                        Message = message,
-                    });
-                }
-            }
-            else
+
+            _pendingMessageQueue.Enqueue(new InputLogEvent
```

Comment:
I still think we need the protection of exceeding max queue size. If the monitor thread dies for some reason this queue would grow boundless till the process ran out of memory.

Reply:
I believe this AddMessage method is the most performance-critical part of the logger because it does not run in the background, and the caller actually waits for it to finish. 

I would rather ensure that the polling thread does not die. How about this?
```csharp
        public void StartMonitor()
        {
            _cancelStartSource = new CancellationTokenSource();
            Task.Run(async () =>
            {
                try
                {
                    await Monitor(_cancelStartSource.Token).ConfigureAwait(false);
                }
                catch
                {
                    StartMonitor();
                }                
            });
        }
```

Reply:
I don't believe the if check is going to affect performance. With this code if something is causing the monitor to keep failing it would recursively use up all of the Tasks in the task pool. To keep the monitor alive I'd recommend using a loop instead of recursion so you can reuse the same Task. Also make sure to not bury Canceled task exceptions.

Reply:
_"I don't believe the if check is going to affect performance."_
Maybe not, but I'm sure those checks took part of the original concurrency issue.

_"To keep the monitor alive I'd recommend using a loop instead of recursion so you can reuse the same Task."_
Thanks, I'll try that.

_"Also make sure to not bury Canceled task exceptions."_
True! Thanks.